### DISCUSSION TITLE
Designer - like TerminalGuiDesigner?

### DISCUSSION THREAD

**Discussion by matho53** [ratatui/ratatui#1354] [Ideas]
What's the projects thoughts towards a Designer tool like TerminalGuiDesigner -> https://github.com/gui-cs/TerminalGuiDesigner
for https://github.com/gui-cs/Terminal.Gui?

**Comment by orhun** [comment]
Honestly I'd love to have something like this. Not built-in to the Ratatui itself but something like a `ratatui-designer` maybe.

There might be some technical challenges to map the actual design to the Layout code but at the end of the day this will further ease the process of building TUIs.

Just for reference:

- https://github.com/gosub-io/poc-taffy
- https://github.com/joshka/tui-brulee

**Reply by matho53** [reply]
Thank you didn't know/notice those projects :) will have a play...
Agree anything to help user's (me) in building TUI's is a win!

**Reply by orhun** [reply]
Those are some layout experimentations that might come in handy in some cases - not sure if it is fully related to what we are trying to build here. But still, they might be inspirational :)

**Comment by joshka** [comment]
Sounds like a good idea to me.

Also take a look at these layout examples in the Ratatui repo for some related work:
- https://github.com/ratatui/ratatui/blob/main/examples/README.md#constraint-explorer
- https://github.com/ratatui/ratatui/blob/main/examples/README.md#constraints
- https://github.com/ratatui/ratatui/blob/main/examples/README.md#flex
